core/genesis.go

```diff
@@ -240,12 +240,21 @@ func SetupGenesisBlock(db ethdb.Database, genesis *Genesis) (*params.ChainConfig
 
 	// Check config compatibility and write the config. Compatibility errors
 	// are returned to the caller unless we're already at block zero.
-	headBlock := rawdb.ReadHeadBlock(db)
-	if headBlock == nil {
-		return newcfg, fmt.Errorf("missing head block")
+	// headBlock := rawdb.ReadHeadBlock(db)
+	// if headBlock == nil {
+	// 	return newcfg, fmt.Errorf("missing head block")
+	// }
+
+	// we use last accepted block for cfg compatibility check. Note this allows
+	// the node to continue making progress if it had halted previously due to
+	// a missing upgrade configuration.
+	lastBlock := ReadBlockByHash(db, lastAcceptedHash)
+	if lastBlock == nil {
+		return newcfg, fmt.Errorf("missing last accepted block")
 	}
```

Comment:
What's returned here if we are starting up for the first time?

Reply:
genesis hash https://github.com/ava-labs/subnet-evm/blob/master/plugin/evm/vm.go#L843-L845

Reply:
I see, forgot about this. Can we add a comment that mentions this requirement here?

Reply:
done